source/particle/generator/ascii_file.cc

```diff
@@ -0,0 +1,173 @@
+/*
+  Copyright (C) 2015 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#include <aspect/particle/generator/ascii_file.h>
+
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Generator
+    {
+      template <int dim>
+      AsciiFile<dim>::AsciiFile() {}
+
+      template <int dim>
+      std::multimap<types::LevelInd, Particle<dim> >
+      AsciiFile<dim>::generate_particles()
+      {
+        const std::string filename = data_directory+data_filename;
+        std::ifstream in(filename.c_str(), std::ios::in);
+        AssertThrow (in,
+                     ExcMessage (std::string("Couldn't open data file <"
+                                             +
+                                             filename
+                                             +
+                                             ">.")));
+
+        // Read header lines
+        while (in.peek() == '#')
+          {
+            std::string temp;
+            getline(in,temp);
+          }
+
+        // Read data lines
+        types::particle_index id = 0;
+        Point<dim> coordinates;
+        std::multimap<types::LevelInd, Particle<dim> > particles;
+
+        if (dim == 2)
+          while (in >> coordinates[0] >> coordinates[1])
+            {
+              particles.insert(this->generate_particle(coordinates,id));
+              id++;
```

Comment:
Here (and in the 3d case a few lines below) you create particles whether they are on a local cell or not. If I read `generate_particle` correctly, this will not work in parallel. But you should be able to recover if you simply did
```
  try {
    particles.insert (this->generate_particle(coordinates,id));
 } 
 catch (...) 
 {}
 ++id;
```

It would probably be useful to declare an named exception class with `DeclExceptionMsg` (instead of `ExcMessage`) that `generate_particle()` could throw if a point is not locally owned. In the catch above, you could then catch only this one exception type.